framework/core/src/Post/Command/PostReplyHandler.php

```diff
@@ -78,10 +79,16 @@ public function handle(PostReply $command)
             $actor->assertCan('reply', $discussion);
         }
 
+        // Get the post type from the query data. When none is provided, we'll
+        // assume it's a comment.
+        $type = Arr::get($command->data, 'attributes.type', 'comment');
+        // Get the post model class that should be used to create it.
+        $postModel = Post::getModels()[$type];
+
         // Create a new Post entity, persist it, and dispatch domain events.
         // Before persistence, though, fire an event to give plugins an
         // opportunity to alter the post entity based on data in the command.
-        $post = CommentPost::reply(
+        $post = $postModel::reply(
```

Comment:
Not all post models will have this interface for the `reply` method, so this isn't structurally sound.